lib/src/view/engine/engine_lines.dart

```diff
@@ -0,0 +1,174 @@
+import 'package:lichess_mobile/src/widgets/buttons.dart';
+
+class EngineLines extends ConsumerWidget {
+  const EngineLines({
+    required this.onTapMove,
+    required this.clientEval,
+    required this.isGameOver,
+    required this.isLandscape,
+  });
+  final void Function(NormalMove move) onTapMove;
+  final ClientEval? clientEval;
+  final bool isGameOver;
+  final bool isLandscape;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final numEvalLines = ref.watch(
+      analysisPreferencesProvider.select(
+        (p) => p.numEvalLines,
+      ),
+    );
+    final engineEval = ref.watch(engineEvaluationProvider).eval;
+    final eval = engineEval ?? clientEval;
+
+    final emptyLines = List.filled(
+      numEvalLines,
+      const Engineline.empty(),
+    );
+
+    final content = isGameOver
+        ? emptyLines
+        : (eval != null
+            ? eval.pvs
+                .take(numEvalLines)
+                .map(
+                  (pv) => Engineline(onTapMove, eval.position, pv),
+                )
+                .toList()
+            : emptyLines);
+
+    if (content.length < numEvalLines) {
+      final padding = List.filled(
+        numEvalLines - content.length,
+        const Engineline.empty(),
+      );
+      content.addAll(padding);
+    }
+
+    return Padding(
+      padding: EdgeInsets.symmetric(
+        vertical: isLandscape ? kTabletBoardTableSidePadding : 0.0,
+        horizontal: isLandscape ? kTabletBoardTableSidePadding : 0.0,
+      ),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        mainAxisAlignment: MainAxisAlignment.start,
+        children: content,
+      ),
+    );
+  }
+}
+
+class Engineline extends ConsumerWidget {
+  const Engineline(
+    this.onTapMove,
+    this.fromPosition,
+    this.pvData,
+  );
+
+  const Engineline.empty()
+      : onTapMove = null,
+        pvData = const PvData(moves: IListConst([])),
+        fromPosition = Chess.initial;
+
+  final void Function(NormalMove move)? onTapMove;
+  final Position fromPosition;
+  final PvData pvData;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    if (pvData.moves.isEmpty) {
+      return const SizedBox(
+        height: kEvalGaugeSize,
+        child: SizedBox.shrink(),
+      );
+    }
+
+    final pieceNotation = ref.watch(pieceNotationProvider).maybeWhen(
+          data: (value) => value,
+          orElse: () => defaultAccountPreferences.pieceNotation,
+        );
+
+    final lineBuffer = StringBuffer();
+    int ply = fromPosition.ply + 1;
+    pvData.sanMoves(fromPosition).forEachIndexed((i, s) {
+      lineBuffer.write(
+        ply.isOdd
+            ? '${(ply / 2).ceil()}. $s '
+            : i == 0
+                ? '${(ply / 2).ceil()}... $s '
+                : '$s ',
+      );
+      ply += 1;
+    });
+
+    final brightness = ref.watch(currentBrightnessProvider);
```

Comment:
We should prefer the flutter way of getting brightness, this refactoring is the occasion to do it:
```suggestion
    final brightness = Theme.of(context).brightness;
```

Reply:
I shall add that we need this provider because it is using the user preferences, but ideally it should be a private provider that only controls the brightness set in the `MaterialApp` widget.